odkx-src/injected-interfaces.rst

```diff
@@ -733,3 +786,520 @@ removeFirstQueuedAction
+Callback functions are registered to handle the success or failure of the update operation.
+
+deleteRow
+~~~~~~~~~~
+
+  **Parameters**:
+
+  - tableId: The ID of the table from which the row should be deleted.
+  - columnNameValueMap: A JSON object representing the column name to value mapping for identifying the row to be deleted.
+  - rowId: The ID of the specific row to be deleted.
+  - successCallbackFn: A success callback function that is called when the row deletion operation is successful.
+  - failureCallbackFn: A failure callback function that is called when there is an error in deleting the row.
+
+This function is used to delete a specific row from a table.
+It takes a JSON object ``columnNameValueMap`` that maps column names to their values, and a ``rowId`` to identify the row to be deleted.
+The function uses the ``getOdkDataIf().deleteRow`` method to delete the row based on the provided criteria.
+Callback functions are registered to handle the success or failure of the delete operation.
+
+addRow
+~~~~~~~
+
+  Parameters:
+
+  - tableId: The ID of the table to which the new row should be added.
+  - columnNameValueMap: A JSON object representing the column name to value mapping for the new row.
+  - rowId: The ID for the new row.
+  - successCallbackFn: A success callback function that is called when the row addition operation is successful.
+  - failureCallbackFn: A failure callback function that is called when there is an error in adding the row.
+
+This function is used to add a new row to a table.
+It takes a JSON object ``columnNameValueMap`` that maps column names to their values for the new row.
+The ``rowId`` is used to specify the ID of the new row.
+The function uses the ``getOdkDataIf().addRow`` method to add the new row to the specified table.
+Callback functions are registered to handle the success or failure of the addition operation.
+
+addCheckpoint
+~~~~~~~~~~~~~~
+
+  **Parameters**:
+
+  - tableId: The ID of the table where the checkpoint should be added.
+  - columnNameValueMap: A JSON object representing the column name to value mapping for the checkpoint.
+  - rowId: The ID for the checkpoint.
+  - successCallbackFn: A success callback function that is called when the checkpoint addition operation is successful.
+  - failureCallbackFn: A failure callback function that is called when there is an error in adding the checkpoint.
+
+This function is used to add a checkpoint to a table.
+A checkpoint is a saved state or record of the data at a specific point in time.
+It takes a JSON object ``columnNameValueMap`` that maps column names to their values for the checkpoint.
+The ``rowId`` is used to specify the ID of the checkpoint.
+The function uses the ``getOdkDataIf().addCheckpoint`` method to add the checkpoint to the specified table.
+Callback functions are registered to handle the success or failure of the addition operation.
+
+saveCheckpointAsIncomplete
```

Comment:
For this and next, may want to cross-link to documentation elsewhere on save types